lib/MemoryModel/MemModel.cpp

```diff
@@ -189,7 +189,7 @@ bool SymbolTableInfo::computeGepOffset(const llvm::User *V, LocationSet& ls) {
         // Handling array types, skipe array handling here
         // We treat whole array as one, then we can distinguish different field of an array of struct
         // e.g. s[1].f1 is differet from s[0].f2
-        if(isa<ArrayType>(*gi))
+        if(isa<ArrayType>(gi.getIndexedType()))
```

Comment:
Afternoon (well, it's afternoon here :) ).  So I created a small test case for a bridging iterator that I hope solves this problem.  I created it separately both to allow SVF to move forward with 4.0 compatibility but also since @dtzWill mentioned that he had a project with a similar issue, I tried to make it somewhat generic.  

@rockysui Would you mind looking at:
https://github.com/jcarlson23/bridgedGEPiterator

And seeing if that helps out in the test cases you brought up?  @dtzWill Obviously, if you need changes for that other project feel free to let me know of alter as you see fit.

Let me know how it goes...